app/views/shared/_modal_form.html.erb

```diff
@@ -9,8 +9,8 @@
         </div>
         <%= simple_form_for [@product, @review] do |f| %>
           <div class="modal-body">
-            <%= f.input :rating %>
-            <%= f.input :comment, as: :text %>
+            <%= f.input :rating, input_html: { min: '0', max: '5' }, required: true %>
+            <%= f.input :comment, as: :text, required: true, error: 'This is a custom error message' %>
```

Comment:
Looks like you have `browser_validations = true` in simple_form. We might not need this 'custom error message'. Or else, change the error message to be more specific than this.